### PR TITLE
ARCO-219: logs improvement

### PR DESCRIPTION
**Pull request by arkadiuszos4chain**
| NOTE: Please ignore conflicts with the `main` branch for now

## Description of Changes

* added event ID to the scope context (event ID passed to all microservices by connection interceptors)
* improved logging:
** logged context data
** added "module" attribute to loggers
* enhanced transaction finder:
** improved node connection utilization
** logged errors from transaction sources in the transaction finder
** cached results in memory

## Linked Issues / Tickets

Reference any related issues or tickets, e.g. "Closes #123".

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [x] I have added new unit tests
- [x] All tests pass locally
- [x] I have tested manually in my local environment

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

**Review by kuba-4chain — APPROVED** [review]

File: internal/grpc_opts/grpc_opts.go [hunk 1]
```diff
@@ -63,6 +67,18 @@ func GetGRPCServerOpts(logger *slog.Logger, prometheusEndpoint string, grpcMessa
 	chainUnaryInterceptors = append(chainUnaryInterceptors, // Order matters e.g. tracing interceptor have to create span first for the later exemplars to work.
 		recovery.UnaryServerInterceptor(recovery.WithRecoveryHandler(grpcPanicRecoveryHandler)))
 
+	// decore context with event ID
```
Comment (boecklim) [inline internal/grpc_opts/grpc_opts.go:70]: nit: decorate

File: pkg/api/handler/internal/TxFinder/tx_finder.go [hunk 2]
```diff
@@ -87,9 +103,10 @@ func (f Finder) GetRawTxs(ctx context.Context, source validator.FindSourceFlag,
 
 	// at last try the WoC
 	if source.Has(validator.SourceWoC) && len(ids) > 0 {
-		wocTxs, _ := f.w.GetRawTxs(ctx, ids)
+		wocTxs, wocErr := f.w.GetRawTxs(ctx, ids)
 		for _, wTx := range wocTxs {
 			if wTx.Error != "" {
+				wocErr = errors.Join(wocErr, fmt.Errorf("returned error data tx %s: %s", wTx.TxID, wTx.Error))
 				continue
```
Comment (boecklim) [inline pkg/api/handler/internal/TxFinder/tx_finder.go:110]: If `wocTxs` is nil or empty, then we'd never see the `wocErr`.
How about this?
```go
		wocTxs, wocErr := f.w.GetRawTxs(ctx, ids)
		if wocErr != nil {
			f.l.WarnContext(ctx, "failed to get raw txs from WoC", slog.String("err", wocErr.Error()))
		}
}
```

Comment (boecklim) [reply pkg/api/handler/internal/TxFinder/tx_finder.go:110]: I see that is already covered on l 123 - 125

**Review by boecklim — APPROVED** [review]

File: pkg/api/handler/internal/TxFinder/cached_tx_finder.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/bitcoin-sv/arc/config"
+	"github.com/bitcoin-sv/arc/internal/validator"
+	"github.com/bitcoin-sv/arc/internal/woc_client"
+	"github.com/bitcoin-sv/arc/pkg/metamorph"
+	"github.com/patrickmn/go-cache"
+)
+
+const (
+	cacheExpiration = 2 * time.Second
+	cacheCleanup    = 5 * time.Second
```
Comment (boecklim) [inline pkg/api/handler/internal/TxFinder/cached_tx_finder.go:17]: Wouldn't the cache be more effective if these time durations are a bit longer?

File: internal/grpc_opts/grpc_opts.go (hunk 1 above)
Comment (arkadiuszos4chain) [reply internal/grpc_opts/grpc_opts.go]: done

**Pull request merged**